flytekit/core/python_function_task.py

```diff
@@ -351,18 +351,23 @@ def dynamic_execute(self, task_function: Callable, **kwargs) -> Any:
     def _write_decks(self, native_inputs, native_outputs_as_map, ctx, new_user_params):
         if self._disable_deck is False:
             from flytekit.deck import Deck
+            from flytekit.deck.deck import DeckFields
             from flytekit.deck.renderer import PythonDependencyRenderer
 
             # These errors are raised if the source code can not be retrieved
             with suppress(OSError, TypeError):
                 source_code = inspect.getsource(self._task_function)
                 from flytekit.deck.renderer import SourceCodeRenderer
 
-                source_code_deck = Deck("Source Code")
+                source_code_deck = Deck(
+                    DeckFields.SOURCE_CODE.value, auto_add_to_deck=DeckFields.SOURCE_CODE in self.decks
+                )
                 renderer = SourceCodeRenderer()
                 source_code_deck.append(renderer.to_html(source_code))
 
-            python_dependencies_deck = Deck("Dependencies")
+            python_dependencies_deck = Deck(
```

Comment:
Same here regarding not needing to creating the deck, if it is not added to the deck.

```python
if DeckFields.DEPENDENCIES in self.decks:
    python_dependencies_deck = Deck("Dependencies")
```